integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -64,6 +70,20 @@ public static void checkReversibleEncryption() {
         assertEquals(PLAINTEXT, new String(decode(TOPIC_1, ByteBuffer.wrap(TOPIC_1_CIPHERTEXT)).array(), StandardCharsets.UTF_8));
         assertArrayEquals(TOPIC_2_CIPHERTEXT, encode(TOPIC_2, ByteBuffer.wrap(PLAINTEXT.getBytes(StandardCharsets.UTF_8))).array());
         assertEquals(PLAINTEXT, new String(decode(TOPIC_2, ByteBuffer.wrap(TOPIC_2_CIPHERTEXT)).array(), StandardCharsets.UTF_8));
+
+        final LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+        final Configuration config = ctx.getConfiguration();
+        appender = (NettyLeakLogAppender) config.getAppenders().get("NettyLeakLogAppender");
```

Comment:
Urgh! This is both ingenious and nasty. 

I think it still suffers from the fact the leak detector can only work on GC cycles, it detects things being freed with a positive reference count. Maybe after each should also request a GC cycle.

edit: I see there is an attempt to force GC in the test itself.

Reply:
Yeahp a bit painful, it can possibly cause a non-deterministic failure in an unrelated test in KrpcFilterIT, but the leak trace should point you in the right direction. Better than a silent leak :(